VisualPinball.Unity/VisualPinball.Unity/VPT/Table/TableApi.cs

```diff
@@ -45,7 +45,8 @@ public class TableApi : IApi
 		private readonly Dictionary<string, TeleporterApi> _teleportersByName = new Dictionary<string, TeleporterApi>();
 		private readonly Dictionary<string, TriggerApi> _triggersByName = new Dictionary<string, TriggerApi>();
 		private readonly Dictionary<string, TroughApi> _troughsByName = new Dictionary<string, TroughApi>();
-
+		private readonly Dictionary<string, CollisionSwitchApi> _collisionSwitchesByName = new Dictionary<string, CollisionSwitchApi>();

```

Comment:
Why not `IApiSwitch` instead of `CollisionSwitchApi`?

Reply:
Okay, so you want `switchesByName`, `switchesByComponent`? 

```
public IApiSwitch Switchable(string name) => _switchesByName[name];
public IApiSwitch Switchable(MonoBehaviour component) => _switchesByComponent[component];
```

Reply:
Yeah, I think that would be more future-proof. You don't need to cast it back in the collider switch API, do you?

Reply:
Looks like I will have to make `IApiSwitch` public instead of internal.

I get inconsistent accessibility.

Reply:
Yup, possible, go ahead :)

Reply:
So I tried to use `IApiSwitch` instead,  but then this would fail:

```
			var nameDict = GetNameDictionary<T>();
			var compDict = GetComponentDictionary<T>();

			nameDict[component.name] = api;
			compDict[component] = api;
```

I didn't want to put something like `if (!(api is CollisionSwitchApi) ` around it. 

Feel free to change this if you want.

Reply:
There already was `_switchablesByComponent` and `_switchablesByName`, using those. Needed to remove the exception because we now save references to switchables where we don't have a reference to the actual typed object.